pkg/reconciler/instances/ory/test/ory_integration_test.go

```diff
@@ -78,3 +31,74 @@ func TestOryIntegration(t *testing.T) {
 		require.Equal(t, true, ready)
 	}
 }
+
+func TestOryIntegrationProduction(t *testing.T) {
+	skipTestIfDisabled(t)
+
+	if !isProductionProfile() {
+		t.Skipf("Integration tests disabled: skipping parts of test case '%s'", t.Name())
+	}
+
+	setup := newOryTest(t)
+	defer setup.contextCancel()
+
+	t.Run("ory-hydra on production profile is deployed", func(t *testing.T) {
+		name := "ory-hydra"
+		hpa := setup.getHpa(t, name)
+
+		require.GreaterOrEqual(t, int32(1), hpa.Status.CurrentReplicas)
+		require.Equal(t, int32(3), hpa.Spec.MaxReplicas)
+		setup.logger.Infof("HorizontalPodAutoscaler %v is deployed", hpa.Name)
+	})
+
+	t.Run("ory-oathkeeper on production profile is deployed", func(t *testing.T) {
+		name := "ory-oathkeeper"
+		hpa := setup.getHpa(t, name)
+
+		require.GreaterOrEqual(t, int32(3), hpa.Status.CurrentReplicas)
+		require.Equal(t, int32(10), hpa.Spec.MaxReplicas)
+		setup.logger.Infof("HorizontalPodAutoscaler %v is deployed", hpa.Name)
+	})
+
+	t.Run("ory-postgresql on production profile is deployed", func(t *testing.T) {
+		name := "ory-postgresql"
+		sts := setup.getSts(t, name)
+
+		require.Equal(t, int32(1), sts.Status.Replicas)
+		require.Equal(t, int32(1), sts.Status.ReadyReplicas)
+		setup.logger.Infof("StatefulSet %v is deployed", sts.Name)
+	})
+
+	t.Run("ory secrets on production profile are deployed", func(t *testing.T) {
+		jwksName := "ory-oathkeeper-jwks-secret"
```

Comment:
What about introducing a common function for secrets verification:
```go
func ensureSecretIsDeployed(t *testing.T, name string, test *oryTest) (error, bool) {
    secret := setup.getSecret(t, name)
    require.NotNil(t, secret.Data)
    test.logger.Infof("Secret %v is deployed", secret.Name)
}
```

Reply:
we can do so :)